app/javascript/crayons/Button/Button.jsx

```diff
@@ -0,0 +1,110 @@
+import { h } from 'preact';
+import PropTypes from 'prop-types';
+import { defaultChildrenPropTypes } from '../../src/components/common-prop-types';
+
+function getAdditionalClassNames({
+  variant,
+  className,
+  icon,
+  disabled,
+  children,
+}) {
+  let additionalClassNames = '';
+
+  if (variant && variant.length > 0 && variant !== 'primary') {
+    additionalClassNames += ` crayons-btn--${variant}`;
+  }
+
+  if (icon) {
+    additionalClassNames +=
+      children.length > 0
+        ? ' crayons-btn--icon-left'
+        : ' crayons-btn--icon-alone';
+  }
+
+  if (disabled) {
+    additionalClassNames += ' crayons-btn--disabled';
+  }
+
+  if (className && className.length > 0) {
+    additionalClassNames += ` ${className}`;
+  }
+
+  return additionalClassNames;
+}
+
+export const Button = ({
+  children,
+  variant = 'primary',
+  as = 'button',
+  className,
+  icon,
+  url,
+  buttonType,
+  disabled,
+  onClick,
+  onMouseOver,
+  onMouseOut,
+  onFocus,
+  onBlur,
```

Comment:
quite a lot of params :D

Reply:
These are all properties of the props parameter, so technically it's one param. 😄 They are all potentially required. It's not that super uncommon. Think of an HTML element and all the potential attributes they can have. 😉